File: Clojure/problem020.clj
#! /usr/bin/env clojure
; Problem 20 - Project Euler
; http://projecteuler.net/index.php?section=problems&id=020

(require '[clojure.string :refer [split]])

(->> (-> (->> (range 1 (inc 100))
              (reduce *'))
         (str)
         (split #""))
     (map #(Integer/parseInt %))
     (reduce +)
     (println))

File: TypeScript/problem020.ts
/*
 * #20 Factorial digit sum - Project Euler
 * http://projecteuler.net/index.php?section=problems&id=20
 */
let x = 1n;
for (let i = 100n; i > 0; i--) {
    x *= i;
}

const ans = [...x.toString()]
    .map((c) => parseInt(c))
    .reduce((acc, x) => (isNaN(x) ? acc : acc + x));
console.log(ans);
